vertx-sql-client/src/main/java/io/vertx/sqlclient/SqlClient.java

```diff
@@ -54,6 +54,13 @@
    */
   PreparedQuery<RowSet<Row>> preparedQuery(String sql, PrepareOptions options);
 
+  /**
+   * Like {@link #preparedQuery(String, PrepareOptions)} with options in JSON format.
+   */
+  default PreparedQuery<RowSet<Row>> preparedQuery(String sql, JsonObject options) {
```

Comment:
I prefer to avoid this, if users need they can create a PrepareOptions from a JsonObject

Reply:
If users create generic `PrepareOptions` from `JsonObject`, the Oracle specific options will be lost, won't they?

Reply:
What we could do instead in Oracle Client is what the various `MetricFactory` do with `MetricsOptions`, for example: https://github.com/vert-x3/vertx-micrometer-metrics/blob/103c5f024d3f36e40a704254e3eb8d28dfec0bd5/src/main/java/io/vertx/micrometer/impl/VertxMetricsFactoryImpl.java#L40-L45

Reply:
yes that would work nicely I think

Reply:
>also sometimes what @pmlopes did is that the data object extends the JsonObject directly

@vietj @pmlopes any specific example in mind?